### PR TITLE
update for sundials 5.1

### PR DESCRIPTION
**Pull request by jd-lara**
@ViralBShah This builder will likely fail because I haven't been able to run the wizard fully because of https://github.com/LLNL/sundials/issues/23

### CODE REVIEW

File: S/Sundials/build_tarballs.jl [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
+# patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
 
 CMAKE_FLAGS=(-DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}")
 CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release -DEXAMPLES_ENABLE_C=OFF)
 CMAKE_FLAGS+=(-DKLU_ENABLE=ON -DKLU_INCLUDE_DIR="$prefix/include" -DKLU_LIBRARY_DIR="$libdir")
-CMAKE_FLAGS+=(-DBLAS_ENABLE=OFF -DLAPACK_ENABLE=OFF)
+CMAKE_FLAGS+=(-DLAPACK_ENABLE=ON)
 
-#if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
-#    patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
```
Comment (ViralBShah) [inline S/Sundials/build_tarballs.jl]: We still should need this patch to mangle the BLAS and LAPACK names correctly to use our `64_` prefixes.

File: S/Sundials/build_tarballs.jl [hunk 2]
```diff
@@ -1,31 +1,29 @@
 using BinaryBuilder
 
 name = "Sundials"
-version = v"3.1.1"
+version = v"5.1.0"
 
 # Collection of sources required to build SundialsBuilder
 sources = [
     "https://github.com/LLNL/sundials/archive/v$(version).tar.gz" =>
     "d03fca24d7adbfe52e05bd4e61340251e1cf56927540cc77a1ca817716091166",
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]:
```suggestion
    "101be83221f9a0ab185ecce04d003ba38660cc71eb81b8a7cf96d1cc08b3d7f9",
```

File: S/Sundials/build_tarballs.jl (hunk 1 above)
Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: That's now done in a different file `config/SundialsFortran`. I have a patch file but didn't add it yet.

Comment (ViralBShah) [reply S/Sundials/build_tarballs.jl]: Could you add it? That way we can clone this PR and see how it goes.

Comment (ViralBShah) [reply S/Sundials/build_tarballs.jl]: In the meanwhile let's see what we need for the 32-bit builders to go through.

Comment (giordano) [reply S/Sundials/build_tarballs.jl]: It looks like most of the platforms that don't have 64-bit integers OpenBLAS are going well.

Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: we need to check that BLAS and LAPACK are detected as operational by Sundials.

Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: @giordano https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1117&view=logs&j=8e1f3fbd-1eec-58ce-53a3-f5a5154dbbe1&t=753643ea-faac-5ab7-7043-7b47880b417b&l=1717

Comment (giordano) [reply S/Sundials/build_tarballs.jl]: In #419 I'm `grep`ping `config.log` to make sure that some libraries we need are correctly identified.  You can do something similar, if you find something usable in the same in the gazillions of files that CMake generates.

**Comment by jd-lara** [comment]
It looks to me like the tests for LAPACK require finding libgfortran when linking with C linkers,
and GCC doesn't automatically add that search path. So rpath-link is needed but I am not sure how to pass it into the test.

File: S/Sundials/build_tarballs.jl (hunk 1 above)
Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: > We still should need this patch to mangle the BLAS and LAPACK names correctly to use our `64_` prefixes.

@ViralBShah added in da9aa31 but can't tell why is causing the pipeline to fail

**Comment by giordano** [comment]
> It looks to me like the tests for LAPACK require finding libgfortran when linking with C linkers,
and GCC doesn't automatically add that search path. So rpath-link is needed but I am not sure how to pass it into the test.

I can have a look in the next days, unless you figure out yourself. You can try for example to do
```
export CFLAGS="-Wl,-rpath-link,/opt/${target}/${target}/sys-root/lib64" 
```
before running cmake

**Comment by jd-lara** [comment]
@giordano thanks I added it but get the same result. 

```
Appending Fortran release flags
-- Determining Fortran name-mangling scheme... FAILED
-- Looking for LAPACK libraries... OK
-- Checking if LAPACK works... FAILED
CMake Warning at config/SundialsCMakeMacros.cmake:83 (message):
  ------------------------------------------------------------------------

  WARNING: LAPACK not functional

  Blas/Lapack support will not be provided

  ------------------------------------------------------------------------
```

**Comment by giordano** [comment]
I'm not at the computer to see what happens, I'll be able to get back to this in a couple of days

**Comment by ViralBShah** [comment]
Are there other Fortran libs being called except blas and lapack? Perhaps those are getting name mangled too?

I am on my phone so unable to dive in on details, which I should be able to in a couple of hours.

**Comment by jd-lara** [comment]
@ViralBShah I am not sure but it seems reasonable. 

I uploaded some extra output here https://github.com/LLNL/sundials/issues/23#issuecomment-577328132 for your reference about the LapackTest that Sundials runs.

File: S/Sundials/build_tarballs.jl [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
-#fi
+CMAKE_FLAGS+=(-DLAPACK_ENABLE=ON)
+
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    # patch -p0 < $WORKSPACE/srcdir/patches/Sundials_fortran.patch
+    # export CFLAGS="-Wl,-rpath-link,/opt/${target}/${target}/sys-root/lib64"
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
+else
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
+fi
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]: This seems to work for me to enable support for LAPACK
```sh
if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_Fortran.patch
    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
else
    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
fi
export CFLAGS="-lgfortran -lquadmath"
```

File: S/Sundials/build_tarballs.jl [hunk 4]
```diff
@@ -42,26 +43,7 @@ fi
 # We attempt to build for all defined platforms
 platforms = supported_platforms()
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]:
```suggestion
 platforms = expand_gfortran_versions(platforms)
```

File: S/Sundials/build_tarballs.jl [hunk 5]
```diff
@@ -42,26 +43,7 @@ fi
...
-    LibraryProduct("libsundials_sunlinsolpcg", :libsundials_sunlinsolpcg),
-    LibraryProduct("libsundials_sunlinsolsptfqmr", :libsundials_sunlinsolsptfqmr),
-    LibraryProduct("libsundials_sunlinsolklu", :libsundials_sunlinsolklu),
-    LibraryProduct("libsundials_sunmatrixsparse", :libsundials_sunmatrixsparse),
-    LibraryProduct("libsundials_sunlinsolband", :libsundials_sunlinsolband),
-    LibraryProduct("libsundials_sunmatrixband", :libsundials_sunmatrixband),
-    LibraryProduct("libsundials_kinsol", :libsundials_kinsol),
-    LibraryProduct("libsundials_arkode", :libsundials_arkode),
-]
+products = Vector{LibraryProduct}()
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]: Revert this change?  Maybe there are more, I didn't check, but the old list of products seems to work for me.

Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: There should be 2 extra products for the LAPACK LinSolvers. I will run it locally with these changes to see if it works

File: S/Sundials/build_tarballs.jl (hunk 3 above)
Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: I am locally getting this error when adding the CFLAGS export 

```
CMake Error at /usr/share/cmake/Modules/CMakeTestCCompiler.cmake:52 (message):
  The C compiler

    "/opt/bin/x86_64-linux-gnu-gcc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp
    
    Run Build Command:"/usr/bin/make" "cmTC_01425/fast"
    /usr/bin/make -f CMakeFiles/cmTC_01425.dir/build.make CMakeFiles/cmTC_01425.dir/build
    make[1]: Entering directory '/workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_01425.dir/testCCompiler.c.o
    /opt/bin/x86_64-linux-gnu-gcc --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/   -lgfortran -lquadmat    -o CMakeFiles/cmTC_01425.dir/testCCompiler.c.o   -c /workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_01425
    /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_01425.dir/link.txt --verbose=1
    /opt/bin/x86_64-linux-gnu-gcc --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/ -lgfortran -lquadmat     CMakeFiles/cmTC_01425.dir/testCCompiler.c.o  -o cmTC_01425 
    /opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/4.8.5/../../../../x86_64-linux-gnu/bin/ld: cannot find -lquadmat
    collect2: error: ld returned 1 exit status
    make[1]: *** [CMakeFiles/cmTC_01425.dir/build.make:87: cmTC_01425] Error 1
    make[1]: Leaving directory '/workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp'
    make: *** [Makefile:121: cmTC_01425/fast] Error 2
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:28 (project)
```

Comment (giordano) [reply S/Sundials/build_tarballs.jl]: Did you misspell `quadmath`?

Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: Dah... yeah 🤦‍♂ sorry

File: S/Sundials/build_tarballs.jl [hunk 6]
```diff
@@ -1,31 +1,32 @@
...
-#else
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
-#fi
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_fortran.patch
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
+else
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
+fi
+export CFLAGS="-lgfortran -lquadmat"
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]:
```suggestion
export CFLAGS="-lgfortran -lquadmath"
```

Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: resolved in 
e5d19dc

File: S/Sundials/build_tarballs.jl [hunk 7]
```diff
@@ -1,31 +1,32 @@
...
+CMAKE_FLAGS+=(-DLAPACK_ENABLE=ON)
 
-#if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
-#    patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas64_.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
-#else
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
-#fi
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_fortran.patch
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]:
```suggestion
    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_Fortran.patch
```
Not sure why you didn't just copy [my code](https://github.com/JuliaPackaging/Yggdrasil/pull/421#discussion_r370901380) :wink:

Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: yeah, sorry now it should be working for many platforms. 

It still doesn't work for me locally to get the right list of products. I will be back in a couple of hours.

**Comment by jd-lara** [comment]
@giordano success https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1160&view=logs&j=1c7f5fc0-afe1-5529-66be-94547223e4d8&t=bd8b0e9a-0b2e-506d-fa03-703e1087b697&l=1724

File: S/Sundials/build_tarballs.jl (hunk 7 above)
Comment (jd-lara) [reply S/Sundials/build_tarballs.jl]: I added the products manually since the wizard isn't working well for me.

File: S/Sundials/build_tarballs.jl [hunk 8]
```diff
@@ -1,31 +1,32 @@
...
-#else
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
-#fi
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_Fortran.patch
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
+else
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
+fi
+export CFLAGS="-lgfortran -lquadmath"
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]: If I'm not mistaken, it looks like `libquadmath` is available only for x86_64 and i686 platforms, so this should become:
```suggestion
export CFLAGS="-lgfortran"
if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
    export CFLAGS="${CFLAGS} -lquadmath"
fi
```

File: S/Sundials/build_tarballs.jl [hunk 9]
```diff
@@ -41,26 +42,31 @@ fi
...
-    LibraryProduct("libsundials_sunlinsolspgmr", :libsundials_sunlinsolspgmr),
-    LibraryProduct("libsundials_sunlinsolpcg", :libsundials_sunlinsolpcg),
-    LibraryProduct("libsundials_sunlinsolsptfqmr", :libsundials_sunlinsolsptfqmr),
-    LibraryProduct("libsundials_sunlinsolklu", :libsundials_sunlinsolklu),
-    LibraryProduct("libsundials_sunmatrixsparse", :libsundials_sunmatrixsparse),
-    LibraryProduct("libsundials_sunlinsolband", :libsundials_sunlinsolband),
-    LibraryProduct("libsundials_sunmatrixband", :libsundials_sunmatrixband),
-    LibraryProduct("libsundials_kinsol", :libsundials_kinsol),
-    LibraryProduct("libsundials_arkode", :libsundials_arkode),
+products = [LibraryProduct("libsundials_sunlinsolspfgmr", :libsundials_sunlinsolspfgmr),
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]: These are the products I get in my local build for `x86_64-linux-gnu`:
```julia
products = [
    LibraryProduct("libsundials_arkode", :libsundials_arkode),
    LibraryProduct("libsundials_cvode", :libsundials_cvode),
    LibraryProduct("libsundials_cvodes", :libsundials_cvodes),
    LibraryProduct("libsundials_ida", :libsundials_ida),
    LibraryProduct("libsundials_idas", :libsundials_idas),
    LibraryProduct("libsundials_kinsol", :libsundials_kinsol),
    LibraryProduct("libsundials_nvecmanyvector", :libsundials_nvecmanyvector),
    LibraryProduct("libsundials_nvecserial", :libsundials_nvecserial),
    LibraryProduct("libsundials_sunlinsolband", :libsundials_sunlinsolband),
    LibraryProduct("libsundials_sunlinsoldense", :libsundials_sunlinsoldense),
    LibraryProduct("libsundials_sunlinsolklu", :libsundials_sunlinsolklu),
    LibraryProduct("libsundials_sunlinsollapackband", :libsundials_sunlinsollapackband),
    LibraryProduct("libsundials_sunlinsollapackdense", :libsundials_sunlinsollapackdense),
    LibraryProduct("libsundials_sunlinsolpcg", :libsundials_sunlinsolpcg),
    LibraryProduct("libsundials_sunlinsolspbcgs", :libsundials_sunlinsolspbcgs),
    LibraryProduct("libsundials_sunlinsolspfgmr", :libsundials_sunlinsolspfgmr),
    LibraryProduct("libsundials_sunlinsolspgmr", :libsundials_sunlinsolspgmr),
    LibraryProduct("libsundials_sunlinsolsptfqmr", :libsundials_sunlinsolsptfqmr),
    LibraryProduct("libsundials_sunmatrixband", :libsundials_sunmatrixband),
    LibraryProduct("libsundials_sunmatrixdense", :libsundials_sunmatrixdense),
    LibraryProduct("libsundials_sunmatrixsparse", :libsundials_sunmatrixsparse),
    LibraryProduct("libsundials_sunnonlinsolfixedpoint", :libsundials_sunnonlinsolfixedpoint),
    LibraryProduct("libsundials_sunnonlinsolnewton", :libsundials_sunnonlinsolnewton),
]
```
I'd also recommend sorting them alphabetically, it makes it easier to go through them.

File: S/Sundials/build_tarballs.jl (hunk 8 above)
Comment (ViralBShah) [reply S/Sundials/build_tarballs.jl]: Does gfortran have some sort of a flag that tells you which libraries to link to? I wish it did.

**Comment by jd-lara** [comment]
Some of the remaining failures are due to LAPACK not being found in some architectures and others are when make is called with multiprocessor which I have seen before and just needs to be called without the `-j${nproc}`

**Comment by ViralBShah** [comment]
Lapack is always committed into openblas on all platforms

**Comment by jd-lara** [comment]
> Lapack is always committed into openblas on all platforms

What I meant to say is that it isn't being successful finding it in other platforms. Probably because of gfortran.

**Comment by giordano** [comment]
The patch wasn't applied:
```
[18:33:09] Attempting to apply Sundials_windows.patch...
[18:33:09] can't find file to patch at input line 3
[18:33:09] Perhaps you used the wrong -p or --strip option?
[18:33:09] The text leading up to this was:
[18:33:09] --------------------------
[18:33:09] |--- config/FindKLU.cmake.orig
[18:33:09] |+++ config/FindKLU.cmake
[18:33:09] --------------------------
[18:33:09] No file to patch.  Skipping patch.
[18:33:09] 1 out of 1 hunk ignored
[18:33:09] Patch Sundials_windows.patch could not be applied! Reverting...
```
You need to be consistent with the `-p` level.  I feel like `atomic_path` should error in this case...

**Comment by jd-lara** [comment]
@giordano is there an easy way you guys test locally with the patches? I haven't found a way to run a local test and pass the patch files. I end up making them manually every time.

**Comment by giordano** [comment]
> is there an easy way you guys test locally with the patches? I haven't found a way to run a local test and pass the patch files. I end up making them manually every time.

Something like this: https://juliapackaging.github.io/BinaryBuilder.jl/dev/FAQ/#I-love-the-wizard,-but-now-I-want-to-break-free:-can-I-build-the-tarballs-without-it?-1?

**Comment by jd-lara** [comment]
> > is there an easy way you guys test locally with the patches? I haven't found a way to run a local test and pass the patch files. I end up making them manually every time.
> 
> Something like this: https://juliapackaging.github.io/BinaryBuilder.jl/dev/FAQ/#I-love-the-wizard,-but-now-I-want-to-break-free:-can-I-build-the-tarballs-without-it?-1?

This is exactly what I needed, thanks. I just haven't worked with BinaryBuilder that much.

**Comment by jd-lara** [comment]
@giordano I have run out of ideas so far to fix the compile in windows it seems that the problem is in Sundials itself 

https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1180&view=logs&j=e9d720d7-4263-563a-1d0a-7e5c18a07b54&t=143671a3-b00f-58bd-0b2e-1674ee780dca&l=3647

**Comment by giordano** [comment]
Yes, I also think this is an issue in Sundials

**Comment by giordano** [comment]
For PowerPC it looks like `libsundials_sunlinsollapackband` and `libsundials_sunlinsollapackdense` are not built at all.

**Comment by jd-lara** [comment]
> For PowerPC it looks like `libsundials_sunlinsollapackband` and `libsundials_sunlinsollapackdense` are not built at all.

It's because it doesn't find LAPACK, I need to look into why

File: S/Sundials/build_tarballs.jl [hunk 10]
```diff
@@ -1,31 +1,37 @@
...
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_Fortran.patch
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
+else
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
+fi
+export CFLAGS="-lgfortran"
+if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
+    export CFLAGS="${CFLAGS} -lquadmath"
+fi
```
Comment (giordano) [inline S/Sundials/build_tarballs.jl]:
```sh
if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
    export CFLAGS="${CFLAGS} -lquadmath"
elif [[ "${target}" == powerpc64le-* ]]; then
    export CFLAGS="${CFLAGS} -Wl,-rpath-link,/opt/${target}/${target}/lib64"
fi
```
This partially solves the problem with PowerPC, but after this the lapack test fails with
```
/opt/bin/cc -lgfortran -Wl,-rpath-link,/opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/lib64 -O3 -DNDEBUG  -rdynamic CMakeFiles/ltest.dir/ltest.o  -o ltest -Wl,-rpath,/workspace/destdir/lib /workspace/destdir/lib/libopenblas64_.so 
/workspace/destdir/lib/libopenblas64_.so: undefined reference to `GOMP_parallel@GOMP_4.0'
collect2: error: ld returned 1 exit status
```
Indeed
```sh
sandbox:${WORKSPACE}/srcdir/sundials-5.1.0/build/LapackTest # nm /opt/${target}/${target}/lib64/libgomp.so | grep GOMP_parallel00000000000097b0 T GOMP_parallel_end
00000000000078e0 T GOMP_parallel_loop_dynamic_start
0000000000007900 T GOMP_parallel_loop_guided_start
0000000000007920 T GOMP_parallel_loop_runtime_start
00000000000078c0 T GOMP_parallel_loop_static_start
0000000000009bf0 T GOMP_parallel_sections_start
0000000000009730 T GOMP_parallel_start
```
there is no `GOMP_parallel` symbol.  This is a big issue.

Comment (giordano) [reply S/Sundials/build_tarballs.jl]: Ok, after an hour of trial and error I'm pretty confident that this should work for all flavours of PowerPC:
```sh
export CFLAGS="-lgfortran"
if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
    export CFLAGS="${CFLAGS} -lquadmath"
elif [[ "${target}" == powerpc64le-* ]]; then
    export CFLAGS="${CFLAGS} -lgomp -ldl -lm -lpthread -Wl,-rpath-link,/opt/${target}/${target}/lib64"
fi
```
but it needs also to change minimum GCC version, see next message

**Comment by giordano** [comment]
Change the last line to
```julia
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"5")
```
(or click the checkbox to allow maintainers to push to your branch :wink:)

**Comment by jd-lara** [comment]
@giordano thanks for the fixes for PowerPC. I have been trying to replicate the windows error locally but I get stuck at the C compiler test with the error `sorry - this program has been built without plugin support` 

I don't know if the fact that I am using docker in a Mac might be affecting the exact version of the toolchain that is used.

**Comment by giordano** [comment]
> I don't know if the fact that I am using docker in a Mac might be affecting the exact version of the toolchain that is used.

It shouldn't, the point of BinaryBuilder is to provide the same environment to everyone :confused:

I can reproduce the Windows error locally but I'm mildly confident that this is an upstream error

**Comment by giordano** [comment]
The macOS failure is a [`BinaryBuilder.jl` bug](https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/381#issuecomment-578458020), but a fix is coming in https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/630.

**Comment by giordano** [comment]
One downside of doing something like
```
export CFLAGS="-lgfortran" 
```
is that probably all libraries will be linked to libgfortran, even if they don't need it, which I think is @ViralBShah's point in a comment he made above.

I need to play more with some of the CMake variables to see how to properly do this, yesterday I did some experiments which didn't work very well.

**Comment by ViralBShah** [comment]
Closing this and opening a new PR - so that sundials v3 and v5 can be both built during the transition.
